crates/rpc-client/src/client.rs

```diff
@@ -81,6 +81,21 @@ impl<T> RpcClient<T> {
     pub fn get_ref(&self) -> ClientRef<'_, T> {
         &self.0
     }
+
+    /// Sets the poll interval for the client.
+    ///
+    /// Note: This will only set the poll interval for the client if it is the only reference to the
+    /// inner client. If the reference is held by many, then it will not update the poll interval.
+    pub fn with_poll_interval(mut self, poll_interval: Duration) -> Self {
+        let inner = Arc::get_mut(&mut self.0);
+
+        if let Some(inner) = inner {
+            inner.set_poll_interval(poll_interval);
+            self
+        } else {
+            todo!("TBD: Value is already shared. Cannot be mutated");
```

Comment:
we could have the poll interval be an atomic number of seconds instead of a duration?

Reply:
Even if we have it as an atomic number, we would still need to use `Arc::get_mut` since `RpcClientInner` is wrapped in `Arc`.

Reply:
atomics don't require a `&mut` reference to modify